connector/src/main/scala/org/apache/spark/sql/cassandra/execution/CassandraDirectJoinStrategy.scala

```diff
@@ -252,19 +255,25 @@ object CassandraDirectJoinStrategy extends Logging {
     */
     reordered.transform {
       case ProjectExec(projectList, child) =>
+        val attrMap = directJoin.output.map {
+          case attr => attr.exprId -> attr
+        }.toMap
+
         val aliases = projectList.collect {
-          case a @ Alias(child: AttributeReference, _) => (child.toAttribute.exprId, a)
+          case a @ Alias(child, _) =>
+            val newAliasChild = child.transform {
+              case attr: Attribute => attrMap.getOrElse(attr.exprId, attr)
```

Comment:
So this just descends till we get the root correct?

Reply:
Yea, it finds and replaces it by the attribute from direct join node.